lib/mongo/protocol/message.rb

```diff
@@ -226,7 +226,7 @@ def serialize_fields(buffer, max_bson_size = nil)
       # @param buffer [String] Buffer to receive the header
       # @return [String] Serialized header
       def serialize_header(buffer)
-        set_request_id unless @request_id
+        set_request_id unless defined?(@request_id)
```

Comment:
"Yes" for the 1st part and "you decide" for the 2nd one. I don't know when all cases set_request_id is called so I leave it to you.